src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -796,6 +796,60 @@ mongoc_collection_count_with_opts (
    RETURN (ret);
 }
 
+/* --------------------------------------------------------------------------
+ *
+ * _make_aggregate_for_edc --
+ *
+ *       Construct an aggregate pipeline with the following form:
+ *
+ *
+ *       { pipeline: [
+ *           { $collStats: { count: {} } },
+ *           { $group: { _id: 1, n: { $sum: $count } } },
+ *         ]
+ *       }
+ *
+ *--------------------------------------------------------------------------
+ */
+static void
+_make_aggregate_for_edc (const mongoc_collection_t *coll, bson_t *out)
+{
+   bson_t pipeline;
+   bson_t coll_stats_stage;
+   bson_t coll_stats_stage_doc;
+   bson_t group_stage;
+   bson_t group_stage_doc;
+   bson_t sum;
+   bson_t cursor_empty;
+   bson_t empty;
+   const char *keys[] = {"0", "1"};
+   int key = 0;
```

Comment:
Can be dropped if you decide to accept the suggestions below:
```suggestion
```

Reply:
Sounds good! Done.